tests/ceph_ansible/test_ansible_roll_over.py

```diff
@@ -121,24 +80,36 @@ def run(ceph_cluster, **kw):
     # use the provided sample file as main site.yml
     ceph_installer.setup_ansible_site_yml(build, ceph_cluster.containerized)
 
-    ceph_cluster.distribute_all_yml()
-
     # add iscsi setting if it is necessary
     if test_data.get("luns_setting", None) and test_data.get("initiator_setting", None):
         ceph_installer.add_iscsi_settings(test_data)
 
     log.info("Ceph versions " + ceph_installer.get_installed_ceph_versions())
 
+    # Run site-x.yml by default
```

Comment:
Would suggest a dictionary to pick right playbook or CLI options instead of if-else blocks. 
```
3.x:
    'mon': "site.yml"
    'osd': "infrastructure-playbook/add-mon-playbook.yaml"
4.1:
    'mon': "infrastructure-playbook/add-mon-playbook.yaml"
    'osd': "infrastructure-playbook/add-mon-playbook.yaml"
4.2:
    'mon': "infrastructure-playbook/add-mon-playbook.yaml"
    'osd': "infrastructure-playbook/add-mon-playbook.yaml"
```